payu/calendar.py

```diff
@@ -90,6 +90,68 @@ def calculate_leapdays(init_date, final_date):
     return datetime.timedelta(days=leap_days)
 
 
+# TODO: The caltype logic could be simplified if we switched
+# to using just a string as the caltype input. Might require reworking of other
+# functions for consistency.
+def seconds_between_dates(start_date, end_date, caltype_int):
+    """
+    Calculate the number of seconds between two datetime objects
+    with a specified calender type by using cftime datetime objects
+    as intermiaries.
+
+    Parameters
+    ----------
+    start_date: datetime.date
+    end_date: datetime.date
+    caltype: Integer, either GREGORIAN or NOLEAP
+
+    Returns
+    -------
+    seconds: Number of seconds between start_date and end_date.
+    """
+    # Get the cftime string corresponding to the caltype integer
+
+    # TODO: Is it confusing that GREGORIAN means proleptic gregorian?
```

Comment:
Well it's confusing that we use `gregorian` when we mean `proleptic_gregorian` and so have to write little work-arounds like this.